src/sentry/issues/auto_source_code_config/constants.py

```diff
@@ -1,17 +1,28 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
 SUPPORTED_INTEGRATIONS = ["github"]
-# XXX: We may want to change these constants into a configuration object
-# Any new languages should also require updating the stacktraceLink.tsx and repo_trees.py SUPPORTED_EXTENSIONS
-SUPPORTED_LANGUAGES = [
-    "csharp",
-    "go",
-    "javascript",
-    "node",
-    "php",
-    "python",
-    "ruby",
-]
-# Some languages will also process system frames
-PROCESS_ALL_FRAMES: list[str] = ["java"]
-# Extract filename from module and abs_path
-# e.g. com.foo.bar.Baz$handle$1, Baz.kt -> com/foo/bar/Baz.kt
-EXTRACT_FILENAME_FROM_MODULE_AND_ABS_PATH = ["java"]
+
+
+# Any new languages should also require updating the stacktraceLink.tsx
+# The extensions do not need to be exhaustive but only include the ones that show up in stacktraces
+PLATFORMS_CONFIG: dict[str, Mapping[str, Any]] = {
+    "csharp": {"extensions": ["cs"]},
```

Comment:
It is not strictly necessary that all extensions are under the right language.

The only important aspect is that the 21 extensions are included:
https://github.com/getsentry/sentry/blob/83cb8ecc71df0e38bf2808334f2d6dbd87f59300/src/sentry/integrations/source_code_management/repo_trees.py#L16-L38